tests/unit/development/__init__.py

```diff
@@ -0,0 +1,136 @@
+import os
+
+_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+def assert_no_trailing_spaces(s):
+    for line in s.split("\n"):
+        assert line == line.rstrip()
+
+
+def match_lines(val, expected):
+    for val1, exp1 in zip(val.splitlines(), expected.splitlines()):
+        assert val1 == exp1
+
+
+def assert_docstring(docstring):
```

Comment:
@Batalex  this is a direct conversion from the original test right?  I don't like the check line by line, it looks long and error-prone with the long list of strings. 

Can we just check the hash of the content ?  like hashlib.md5(string).hexdigest(), the hardcode part will be a simple digest string.

Reply:
@byronz If you are okay with that, I replaced those functions with `difflib`. I think the hash approch is much harder to debug and equivalently error-prone.
Hardcoding the digest string would indeed be more concise, but we would not have any way of knowing why two hashes are different and this would require an extra step every time the docstring template is changed.
Even if we keep the hard coded docstring and compute the hash at runtime, the difficulty to figure out why the hashes are different remains

I also changed some other utils functions, all `assert` are back in the tests

Reply:
@Batalex  I'm sorry for the late reply, I was on vacation last week. what's the difference of using the difflib vs pytest's own diff mechanism? did you find it more readable?

Reply:
No worries !
The main advantage of using difflib instead of what we had before is that we can now detect multiple errors in one test run. The previous version would fail a test on the first different line regardless of the total number of lines changed whereas here we get a git-like diff with all changes